test/e2e/azure_test.go

```diff
@@ -112,6 +122,272 @@ var _ = Describe("Workoad cluster creation", func() {
 				WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
 				WaitForMachineDeployments:    e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
 			})
+
+			By("creating a Kubernetes client to the management cluster")
+			mgmtConfig, err := clientcmd.BuildConfigFromFlags("", bootstrapClusterProxy.GetKubeconfigPath())
+			Expect(err).NotTo(HaveOccurred())
+			mgmtClientset, err := kubernetes.NewForConfig(mgmtConfig)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("fetching the workload cluster's kubeconfig from the secret")
+			secret, err := mgmtClientset.CoreV1().Secrets(namespace.Name).Get(clusterName+"-kubeconfig", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			kubeconfig := secret.Data["value"]
+			Expect(len(kubeconfig)).To(BeNumerically(">", 0))
+			tmpFile, err := ioutil.TempFile(os.TempDir(), specName)
+			Expect(err).NotTo(HaveOccurred())
+			defer os.Remove(tmpFile.Name())
+			_, err = tmpFile.Write(kubeconfig)
+			Expect(err).NotTo(HaveOccurred())
+			err = tmpFile.Close()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating a Kubernetes client to the workload cluster")
+			config, err := clientcmd.BuildConfigFromFlags("", tmpFile.Name())
+			Expect(err).NotTo(HaveOccurred())
+			clientset, err := kubernetes.NewForConfig(config)
+			Expect(err).NotTo(HaveOccurred())
+
```

Comment:
can't we use the BootstrapClusterProxy to get access to the workload cluster without have to do all the kubeconfig steps here, there is an example in cluster-api repo https://github.com/kubernetes-sigs/cluster-api/blob/baedbb83b160ee134a065d3d9d16094e23d4e1a3/test/e2e/self_hosted.go#L114

Reply:
@cpanato does that apply to networking PR as well?

Reply:
in the net pol we are using the `kubectl` command line. That's why I asked in the meeting, should we support both `clientSet` and `command line` or just one? if just one which one :)

If we say we go for the `clientSet` I will refactor the netpol tests to use that. If we support both we can keep as is for now and maybe do some refator in a follow up to refine the helpers.

@nader-ziada @CecileRobertMichon what are your thoughts?

Reply:
I'm on the fence personally, but I agree that we should pick one approach and use it consistently.

Using `kubectl` has some advantages (its `run` and `exec` commands aren't easily reproduced in Go code, and it might be easier code to read), but it also requires the test to fetch and use the correct binary version(s).

Reply:
Agree, so in the regards we can support both ways, depending of the test, if is just to create and check states and other things we can use the client. If need to run something inside the pod or other things we can use kubectl. In the network pol case it exec things to test the connectivity.

Reply:
I put this topic (client-go vs. `kubectl` for e2e) on the CAPZ office hours discussion for tomorrow.